aibolit/ast_framework/java_class_decomposition.py

```diff
@@ -41,31 +56,54 @@ def is_ast_pattern(class_ast: AST, Pattern) -> bool:
     return len(Pattern().value(class_ast)) > 0
 
 
+def _normalize_strength(strength: Union[DecompositionStrength, str]) -> DecompositionStrength:
+    if isinstance(strength, DecompositionStrength):
+        return strength
+    if isinstance(strength, str):
+        try:
+            return DecompositionStrength(strength.lower())
+        except ValueError as exc:
+            valid_strengths = DecompositionStrength.values()
+            raise ValueError(
+                f'Unsupported decomposition strength: {strength}. '
+                f'Must be one of: {valid_strengths}'
+            ) from exc
+    valid_strengths = DecompositionStrength.values()
+    raise ValueError(
+        f'Unsupported decomposition strength: {strength}. '
+        f'Must be one of: {valid_strengths}'
+    )
+
+
 def decompose_java_class(
         class_ast: AST,
-        strength: str,
+        strength: Union[DecompositionStrength, str] = DecompositionStrength.STRONG,
```

Comment:
@kodsurfer thank you for the PR!

I learnt a lot when reviewing this solution! I never thought of this smooth transition from literal strings to Enum before.

I wonder, however, if this enum solution is any better than a simple `strength: Literal['strong', 'weak']`, considering the task at hand?